### ISSUE TITLE
Allow production build without S3 

### ISSUE THREAD

**Issue by mattkrick** [ParabolInc/action#589]
for the folks building at home, they might use node, not S3.
If those folks try running `npm run build:client-min` they get the following using the default .env.example:
```
ERROR in S3Plugin-RequiredS3UploadOpts: Bucket
Uploading [>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 100% 0.0s

npm ERR! Darwin 16.1.0
```
@jordanh let's tackle this one together

**Comment by jordanh** [comment]
Oh ya, this is a good idea.

A proposal:

- Change `webpack.config.prod.js` to only enable S3 uploading when we have a bucket:

```
  if (process.env.AWS_S3_BUCKET && !process.env.CI) {
    // do not deploy to S3 if running in continuous integration environment:
    deployPlugins.push(new S3Plugin({
```

- Comment out AWS section in .env.example by default

```
## AWS S3
#AWS_ACCESS_KEY_ID="XXXXXXXXXXXXXXXXXXXX"
#AWS_REGION="us-west-2"
#AWS_S3_BUCKET="action-XXXXXXXX"
#AWS_SECRET_ACCESS_KEY="XXXXXXXXXXXXXXXXXXXXXXXX"
```

**Comment by mattkrick** [comment]
Another consideration:
I am writing server code, so i run `npm run build` so i don't have to rebuild the client bundle every time, i just run `npm run start`.
I get this turd: 
![screen shot 2017-01-02 at 6 44 42 pm](https://cloud.githubusercontent.com/assets/5514175/21599071/9a9a240e-d11b-11e6-9cbe-b26c0f1f111c.png)

**Comment by jordanh** [comment]
Oh, well, that's just 💩 

I have two options for you. Feel free to introduce more:

1. I make it so that if no `NODE_ENV` is set, it defaults to `production` so you can run `$ npm run build` to your ❤️ 's content for development purposes if you set `NODE_ENV=development` in your local environment

2. I make an `npm run build-dev` command

**Comment by mattkrick** [comment]
could we use `DEPLOY == true` as the hook?
- build:client is for development purposes
- build:client-min is for staging/production purposes
- build:server just gives us prerender.js so it doesn't matter

i could be wrong & maybe build:server is more complex?

**Comment by jordanh** [comment]
Implemented in `circle-2.0` branch
